test/integration/plugins/nuagevsp/nuageTestCase.py

```diff
@@ -77,85 +117,104 @@ def setUpClass(cls, zone=None):
         cls.test_data["virtual_machine"]["zoneid"] = cls.zone.id
         cls.test_data["virtual_machine"]["template"] = cls.template.id
 
-        # Create service offering
-        cls.service_offering = ServiceOffering.create(cls.api_client,
-                                                      cls.test_data["service_offering"]
-                                                      )
-        cls._cleanup = [cls.service_offering]
-
         # Check if the host hypervisor type is simulator
-        cls.isSimulator = Hypervisor.list(cls.api_client, zoneid=cls.zone.id)[0].name == "Simulator"
+        cls.isSimulator = Hypervisor.list(
+            cls.api_client,
+            zoneid=cls.zone.id)[0].name == "Simulator"
 
         # Get configured Nuage VSP device details
         try:
-            physical_networks = PhysicalNetwork.list(cls.api_client, zoneid=cls.zone.id)
+            physical_networks = PhysicalNetwork.list(
+                cls.api_client,
+                zoneid=cls.zone.id
+            )
             for pn in physical_networks:
                 if pn.isolationmethods == "VSP":
                     cls.vsp_physical_network = pn
                     break
-            cls.nuage_vsp_device = Nuage.list(cls.api_client,
-                                              physicalnetworkid=cls.vsp_physical_network.id
-                                              )[0]
+            cls.nuage_vsp_device = Nuage.list(
+                cls.api_client,
+                physicalnetworkid=cls.vsp_physical_network.id)[0]
             pns = cls.config.zones[0].physical_networks
-            providers = filter(lambda physical_network: "VSP" in physical_network.isolationmethods, pns)[0].providers
-            devices = filter(lambda provider: provider.name == "NuageVsp", providers)[0].devices
+            providers = filter(lambda physical_network:
+                               "VSP" in physical_network.isolationmethods,
+                               pns)[0].providers
+            devices = filter(lambda provider:
+                             provider.name == "NuageVsp",
+                             providers)[0].devices
             cls.nuage_vsp_device.username = devices[0].username
             cls.nuage_vsp_device.password = devices[0].password
             cls.cms_id = cls.nuage_vsp_device.cmsid
         except Exception as e:
             cls.tearDownClass()
-            raise unittest.SkipTest("Warning: Could not get configured Nuage VSP device details - %s" % e)
+            raise unittest.SkipTest("Warning: Could not get configured "
+                                    "Nuage VSP device details - %s" % e)
```

Comment:
If the datacenter, against which the tests are run, is not configured with Nuage VSP
we want to skip the Nuage VSP related tests instead of failing.